rtsmith.h

```diff
@@ -8,8 +8,20 @@
 namespace P4Tools::RTSmith {
 
 struct RtSmithResult {
-    InitialP4RuntimeConfig config;
-    P4RuntimeUpdateSeries updateSeries;
+    InitialConfig config;
+    UpdateSeries updateSeries;
+
+    RtSmithResult(InitialConfig &&config, UpdateSeries &&updateSeries)
```

Comment:
I do not think youi need those. They should be available by default or with `RtSmithResult(RtSmithResult &other) = default;`

Reply:
Just checked about these. Only `RtSmithResult(InitialConfig &&config, UpdateSeries &&updateSeries)` is required.

1. For the first constructor: in `runRtSmith` we return `{{initialConfig, timeSeriesUpdates}}`, which is `std::optional<RtSmithResult>`. We need this constructor to move config/updateSeries into the struct. Otherwise, copy constructor will be called and the type `InitialConfig`(`std::vector<std::unique_ptr<google::protobuf::Message>>`) is not "constructible from value" (from compiler errors). 
2. For the copy constructor. `ASSIGN_OR_RETURN` in `flay_checker.cpp:run` requires `RtSmithResult(const RtSmithResult &other)`. But the default copy constructor doesn't work because of the same issue in 1. But I realized that changing this `ASSIGN_OR_RETURN` to something like `MOVE_OR_RETURN` might be better.
3. The move constructor can be removed.

I pushed an update, could you take a look and lmk if it is better?

Reply:
Interesting, requiring the copy constructor indicates a problem. We should be able to use move assignments here.

Reply:
Oh I realized that changing `auto configResult` to `auto &configResult` should solve the issue of point 2.

Reply:
"all warnings being treated as errors". Changed that to `if (!generateConfig...) return false` instead.